stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -624,10 +616,67 @@ public class ChatClient internal constructor(
             .precondition(relevantPlugins) { onDeleteReactionPrecondition(currentUser) }
     }
 
+    /**
+     * Sends the reaction.
+     * Use [enforceUnique] parameter to specify whether the reaction should replace other reactions added by the current user.
+     * [cid] parameter is being used in side effect functions executed by plugins.
+     * You can skip it if plugins are not being used.
+     *
+     * The call will be retried accordingly to [retryPolicy].
+     *
+     * @see [Plugin]
+     * @see [RetryPolicy]
+     *
+     * @param reaction The [Reaction] to send.
+     * @param enforceUnique Flag to determine whether the reaction should replace other ones added by the current user.
+     * @param cid The full channel id, i.e. "messaging:123" to which the message with reaction belongs.
+     *
+     * @return Executable async [Call] responsible for sending the reaction.
+     */
     @CheckResult
     @JvmOverloads
-    public fun sendReaction(reaction: Reaction, enforceUnique: Boolean = false): Call<Reaction> {
+    public fun sendReaction(reaction: Reaction, enforceUnique: Boolean, cid: String? = null): Call<Reaction> {
+        val relevantPlugins = plugins.filterIsInstance<SendReactionListener>()
+        val currentUser = getCurrentUser()
+
         return api.sendReaction(reaction, enforceUnique)
+            .retry(scope = scope, retryPolicy = retryPolicy)
+            .onErrorReturn(scope) { originalError ->
+                val errorHandler = errorHandlerFactory.getOrCreate()
```

Comment:
Please don't create `errorHandler` inside ChatClient as this class should not be aware of how its dependencies are created. We can add the error handlers just like we do with plugins.